diff --git a/src/llama.cpp b/src/llama.cpp
index ed77ed91..f53e76eb 100644
--- a/src/llama.cpp
+++ b/src/llama.cpp
@@ -8125,7 +8125,7 @@ static struct ggml_tensor * llm_build_kqv(

         cur = ggml_flash_attn_ext(ctx, q, k, v, kq_mask, kq_scale, hparams.f_max_alibi_bias);

-        if (model.arch == LLM_ARCH_PHI2 || model.arch == LLM_ARCH_PHI3 || model.arch == LLM_ARCH_GPTNEOX) {
+        if (model.arch == LLM_ARCH_PHI2 || model.arch == LLM_ARCH_PHI3 || model.arch == LLM_ARCH_GPTNEOX || model.arch == LLM_ARCH_QWEN2) {
             ggml_flash_attn_ext_set_prec(cur, GGML_PREC_F32);
         }

